Pocket/pocket-ios
# Native Collections - Fix navigation for sub collections

## Summary
* This PR fixes a bug that caused navigation to stories not working when navigating to a sub collection and going back to the parent collection.
* It also adds purging orphaned collections and stories when refreshing Home

## References 
* branch name
## Implementation Details
* Update `Space`, add purging logic for `Collection` and `CollectionStory`
* Add `SubscriptionsStack`, a convenience type to store sets of Combine subscriptions on a stack, with `push` and `pop` operations, used to pop unused sets to avoid retaining unnecessary references to `CollectionViewModel`
* Update logic to push collection in `HomeViewController` and `SavesContainerViewController` to use `SubscriptionStack`
* Update logic to push collection in `HomeViewController` to fix the navigation issue mentioned in the description (it was caused by removing all collection subscriptions when pushing a new collection, which caused the parent view controller to not receive navigation status updates anymore)

## Test Steps
1. Build/run
2. Tap "Home"
3. Scroll to find a collection with a sub collection (e.g. "5 Tech Habits to Quit for a Happier Life")
4. Open the collection, then found a sub collection
5. Open the sub collection
6. Go back and make sure that you can navigate to other stories (collection or not)
7. Go back
8. Select any item that is not a collection, then go back 
> [!NOTE]
> this is because an instance of `readableType` is assigned in `HomeViewModel`, and will retain a reference to the latest readable type. If you don't select another readable type, you will still see an instance of `CollectionViewModel`. We will need to change it as well, but since it's a `HomeViewModel` refactor that impacts more than just collections, I consider this outside the scope of the current change, and we will do it in a separate PR.
11. From Xcode, open the memory graph and make sure there are no `CollectionViewModel` instances
12. Restart the debugger
13. Save a collection that has sub collections
14. go to saves, select that collection, then a sub collection then go back to the saves list
15. repeat step 8
16. Go to Home
17. Refresh
18. Open Core Data lab (or other tool to check locally saved entities), and make sure that there are no orphaned collections or stories

## PR Checklist:

## Screenshots
